073-improve-handling-of-CA-renewals-and-replacements-in-client-based-operands.md

```diff
@@ -0,0 +1,210 @@
+   * How many nodes does the operand such as Kafka Connect have.
+     If it is only one replica, the whole operand will be down.
+     With multiple replicas, only one replica should be down.
+   * How big and busy the Kafka cluster is.
+     A small cluster might take few minutes to move to the new CA completely.
+     A big cluster with many nodes that are busy and need more time to re-sync between restarts might take much longer.
+6. After all brokers (and other components such as ZooKeeper, Topic and User Operator, Cruise Control etc.) use the new CA, another rolling update will be done to completely remove the trust to the old CA by removing the earlier `ca-<timestamp>.crt` from the Secret and rolling all components once again.
+   This last rolling update does not impact operands such as Kafka Connect.
+7. The removal of `ca-<timestamp>.crt` from the Secret will cause rolling update of the client-based operand (e.g. Connect).
+   This rolling update will remove the trust to the old CA from the operand as well.
+   This is an additional rolling update that happens when the new API is used.
+
+### Examples
+
+The examples for the operands based on Kafka clients will be updated to use the new field:
+```yaml
+  tls:
+    trustedCertificates:
+      - secretName: my-cluster-cluster-ca-cert
+        extension: crt
+```
+
+This will make sure that when used with a Strimzi based cluster, everything - including CA replacement - will work without any outages and issues.
+
+## Affected projects
+
+This proposal affects only the Strimzi Cluster Operator project.
+
+## Backwards compatibility
+
+There is no impact on backwards compatibility.
+This proposal does not remove any Strimzi APIs.
+It only extends them.
+All existing custom resources will work without any change.
+
+## Rejected alternatives
+
+### Using a regular expression instead of extension
+
+One option considered for this proposal was to use a pattern to specify the matching certificates instead of extension.
+However, when using a pattern / regular expression:
+* The correct value would depend on where is the pattern applied (e.g. in Java, in shell when starting the container image, in some different language in the future).
+* For non-expert users it might be hard to understand the exact format and the different regular expression variants.
+* It would be hard to validate the the pattern / regular expression.
+
+That is why this option was rejected.
+
+### Hardcoding the extension
+
+Another considered option was to not add a new field to specify the `extension` at all in the API and just make the `certificate` field optional.
+When the `certificate` field would not be specified, all items from the Secret with the `.crt` extension would be loaded.
+This option was rejected because in some cases, users might use different extension such as `.pem` for public keys and having it hardcoded might make it not work for them.
```

Comment:
To be honest, I don't think it gives them this impression any more than the current API where they can specify any _filename_ they want. I can of course clarify it in the proposal, but the behavior will be exactly the same as it is today -> it will likely get the container crash-looping if you feed it a wrong format.

Do you have any alternative? We can enforce the extension to be always `crt` (and basically remove the `extension` field and just make the `certificate field optional) and let everyone using a different name rename they certs in the Secret. But it does not completely solve the format issue either as you can still have unsupported format stored as `.crt`.

Reply:
>  I can of course clarify it in the proposal, but the behavior will be exactly the same as it is today

Restating the existing behaviour in this proposal is all I was asking for.